### ISSUE TITLE
[NuGet.org Bug]: Wrong additional computed target frameworks if native dependencies

### ISSUE THREAD

**Issue by mfkl** [NuGet/NuGetGallery#9013]
### Impact

It bothers me. A fix would be nice

### Describe the bug

Recently a new feature on nuget.org shows the TFMs on each package's page.

When implicit or explicit native dependencies are involved, the binding project page may show it supports a platform that the native package does not support, which can be effectively misleading to users.

![image](https://user-images.githubusercontent.com/3928834/154900580-3f340762-eef2-4959-b38d-e7d605960d3c.png)

Example above. In theory the binding project could run on watchOS or Tizen, but the underlying native dependency does not.

### Repro Steps

1. On https://www.nuget.org/packages/LibVLCSharp/
2. Click "Frameworks"
3. Witness targets like watchOS or Tizen, which are unsupported by LibVLC (while the binding could in theory run on those platforms, it would be useless, so this is misleading).

### Expected Behavior

Maybe a way to indicate that the computed TFMs are an indication and not guaranteed in case of underlying native dependencies.

**Comment by joelverhagen** [comment]
Hey @mfkl, thanks for taking the time to report this. I have assigned the issue to the two owners of this new feature. 

Could you help me understand where the native dependency of your package is referenced or stated in the package metadata? One potential challenge here is that perhaps NuGet does not have a way yet to model this sort of native dependency in a way you like. From the description on your package, it seems that you need to install the native dependency separately. This makes it hard for our NuGet.org UI to know that anything is amiss.

**Comment by mfkl** [comment]
> Could you help me understand where the native dependency of your package is referenced or stated in the package metadata? From the description on your package, it seems that you need to install the native dependency separately. This makes it hard for our NuGet.org UI to know that anything is amiss.

Yes, indeed, that is correct. We don't have an explicit package reference between the managed binding and the native library, so I understand how it would be even trickier for NuGet to infer that info.

We could take [LibGit2Sharp](https://www.nuget.org/packages/LibGit2Sharp/) as an example of NuGet package with explicit package reference to the native lib though. It references [LibGit2Sharp.NativeBinaries](https://www.nuget.org/packages/LibGit2Sharp.NativeBinaries/) which does not contain native builds for tvOS, Tizen or watchOS, though `LibGit2Sharp`'s additional computed frameworks shows them as supported.

I understand how this would be a challenge on your side though (essentially matching managed wrappers with native lib package and analysing native build support in the latter to reflect that info in the former). Potentially a niche use case too.

Just thought I'd report it anyway. Feel free to close this.

**Comment by joelverhagen** [comment]
This is a good issue report. Thank you for your feedback. I'd like to leave this open since it's clear to me now that there are multiple different approaches to native dependencies out on NuGet.org and the Frameworks tab isn't handling these approaches perfectly.

Any confusion by a package author (you 😀) will certainly exist for package consumers as well!

**Comment by mfkl** [comment]
> Any particular reason why there's an inclusion of netstandard vs. targeting each individual target framework?

We choose to target netstandard as it provides a basis for general compatibility. dotnet CLI apps on mac/windows/linux use that target, for example, as well as Unity apps on all platforms.

> How might we improve this experience?

1.
> Maybe a way to indicate that the computed TFMs are an indication and not guaranteed in case of underlying native dependencies.

and/or

2.
> matching managed wrappers with native lib package and analysing native build support in the latter to reflect that info in the former
